crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -870,12 +868,52 @@ impl SlidingSync {
         self.rooms.read().unwrap().values().cloned().collect()
     }
 
+    fn prepare_extension_config(&self, pos: Option<&str>) -> ExtensionsConfig {
+        if pos.is_none() {
+            // The pos is `None`, it's either our initial sync or the proxy forgot about us
+            // and sent us an `UnknownPos` error. We need to send out the config for our
+            // extensions.
+            let mut extensions = self.extensions.lock().unwrap().clone().unwrap_or_default();
+
+            // Always enable to-device events and the e2ee-extension on the initial request,
+            // no matter what the caller wants.
```

Comment:
I don't exactly understand the why of this, but I assume it's been agreed upon.

Reply:
It's a stop-gap solution which attempts to prevent clients from disabling end-to-end encryption until we figure what we want to do with the extensions for optimal performance.